pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TopicsConsumerImplTest.java

```diff
@@ -863,7 +863,7 @@ public void testConsumerDistributionInFailoverSubscriptionWhenUpdatePartitions()
                 .messageRouter(new MessageRouter() {
                     @Override
                     public int choosePartition(Message<?> msg, TopicMetadata metadata) {
-                        return Integer.parseInt(msg.getKey()) % metadata.numPartitions();
+                        return msg.hasKey() ? Integer.parseInt(msg.getKey()) % metadata.numPartitions() : 0;
```

Comment:
this change looks unrelated to this patch, please revert

Reply:
I'll add this behavior for authn/authz (please see also https://github.com/apache/pulsar/pull/11570#issuecomment-897463605 ). Elect first partition index by MessageRouter with blank message.
https://github.com/equanz/pulsar/blob/844cca7cf84b76b7d27f12de4dbf36e23cbf29e9/pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java#L87-L88

 If we shouldn't change this test and behavior, 

1. add pseudo message key like `xxx` to blank message
2. handle exceptions in PartitionedProducerImpl
3. select first partition index without MessageRouter (e.g. randomly select)
   - if message router elects partition from part of partitions like singlepartition routing mode, create redundant producer for this change